spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractPollingEndpoint.java

```diff
@@ -310,18 +310,21 @@ private IntegrationResourceHolder bindResourceHolderIfNecessary(String key, Obje
 		if (this.transactionSynchronizationFactory != null && resource != null &&
 				TransactionSynchronizationManager.isActualTransactionActive()) {
 			TransactionSynchronization synchronization = this.transactionSynchronizationFactory.create(resource);
-			TransactionSynchronizationManager.registerSynchronization(synchronization);
```

Comment:
You can simply move this inside the if statement. The instance of test will return false for null.